src/hooks/handleQueryInUrl.ts

```diff
@@ -12,7 +12,8 @@ export function useHandleQueryInUrl() {
   const { page } = usePageContext();
 
   useEffect(() => {
-    const hasQueryUrl = Object.keys(router.query).length > 0;
+    const hasQueryUrl = window.location.search !== "";
```

Comment:
for some reason the nextjs router has [weird behavior when navigating around with queries in the url](https://github.com/vercel/next.js/issues/10875). That caused this `hasQueryUrl` to return true when it wasn't supposed to. Checking the window search params directly solves the issue.